test/sshkit/scp/upload_test.exs

```diff
@@ -0,0 +1,155 @@
+defmodule SSHKit.SCP.UploadTest do
+  use ExUnit.Case, async: true
+  import Mox
+
+  alias SSHKit.SCP.Upload
+  alias SSHKit.SSHMock
+
+  @local "test/fixtures/local_dir"
+  @remote "/home/test/code"
+  @conn %SSHKit.SSH.Connection{}
+  @chan %SSHKit.SSH.Channel{}
```

Comment:
I think you were right yesterday. `@conn` and `@chan` look somewhat odd up here. Maybe it indeed makes sense to create these in a `setup` block and pull them from the test context for the tests that need them?